lib/identity_cache/cached/belongs_to.rb

```diff
@@ -37,28 +37,30 @@ def fetch(records)
 
       def fetch_async(load_strategy, records)
         if reflection.polymorphic?
-          cache_keys_to_associated_ids = {}
+          type_fetcher_to_db_ids_hash = {}
 
           records.each do |owner_record|
             associated_id = owner_record.send(reflection.foreign_key)
             next unless associated_id && !owner_record.instance_variable_defined?(records_variable_name)
-            associated_cache_key = Object.const_get(
+            foreign_type_fetcher = Object.const_get(
               owner_record.send(reflection.foreign_type)
             ).cached_model.cached_primary_index
-            unless cache_keys_to_associated_ids[associated_cache_key]
-              cache_keys_to_associated_ids[associated_cache_key] = {}
-            end
-            cache_keys_to_associated_ids[associated_cache_key][associated_id] = owner_record
+            (type_fetcher_to_db_ids_hash[foreign_type_fetcher] ||= []) << associated_id
           end
 
-          load_strategy.load_batch(cache_keys_to_associated_ids) do |associated_records_by_cache_key|
+          load_strategy.load_batch(type_fetcher_to_db_ids_hash) do |batch_load_result|
             batch_records = []
-            associated_records_by_cache_key.each do |cache_key, associated_records|
-              associated_records.keys.each do |id, associated_record|
-                owner_record = cache_keys_to_associated_ids.fetch(cache_key).fetch(id)
-                batch_records << owner_record
-                write(owner_record, associated_record)
-              end
+
+            records.each do |owner_record|
+              associated_id = owner_record.send(reflection.foreign_key)
```

Comment:
I think we can use `public_send` here, right? Same with below

Reply:
Wouldn't that just mean it would fail if the attribute method were marked private?  I'm not sure what advantage that would provide, considering that this is mean to be internal to the model's implementation.

Reply:
True, IDC is very internal-reaching. I don't see why an attribute would be marked as private, but I suppose it is possible. We seem to use send in similar cases already, so it is fine to keep.